common/src/chain/config/regtest.rs

```
@@ -0,0 +1,430 @@
+    #[error("Hex decoding failed {0}")]
+    HexDecodingFailed(String),
+    #[error("Invalid pool Id {0}")]
+    InvalidPoolId(String),
+    #[error("Invalid staking private key {0}")]
+    InvalidStakingPrivateKey(String),
+    #[error("Invalid VRF private key {0}")]
+    InvalidVRFPrivateKey(String),
+    #[error("Type decoding failed {0}")]
+    TypeDecodingFailed(String),
+    #[error("Unknown staking settings parameter {0}")]
+    UnknownParameter(String),
+}
+
+fn decode_hex<T: serialization::DecodeAll>(
+    hex: &str,
+) -> Result<T, GenesisStakingSettingsInputErrors> {
+    let bytes = Vec::from_hex(hex)
+        .map_err(|_| GenesisStakingSettingsInputErrors::HexDecodingFailed(hex.to_string()))?;
+    <T as serialization::DecodeAll>::decode_all(&mut bytes.as_slice())
+        .map_err(|_| GenesisStakingSettingsInputErrors::TypeDecodingFailed(hex.to_string()))
+}
+
+impl GenesisStakingSettings {
+    pub fn new(
+        pool_id: PoolId,
+        stake_private_key: PrivateKey,
+        vrf_private_key: VRFPrivateKey,
+    ) -> Self {
+        Self {
+            pool_id,
+            stake_private_key,
+            vrf_private_key,
+        }
+    }
+
+    pub fn pool_id(&self) -> PoolId {
+        self.pool_id
+    }
+
+    pub fn stake_private_key(&self) -> PrivateKey {
+        self.stake_private_key.clone()
+    }
+
+    pub fn vrf_private_key(&self) -> VRFPrivateKey {
+        self.vrf_private_key.clone()
+    }
+}
+
+impl Default for GenesisStakingSettings {
+    fn default() -> Self {
+        Self::from_str("").expect("Default value")
+    }
+}
+
+impl FromStr for GenesisStakingSettings {
+    type Err = GenesisStakingSettingsInputErrors;
+
+    fn from_str(settings: &str) -> Result<Self, Self::Err> {
+        let mut settings_parts = settings
+            .split(',')
+            .filter_map(|s| s.split_once(':'))
+            .map(|(k, v)| (k.trim(), v.trim()))
+            .collect::<HashMap<&str, &str>>();
```

The `filter_map` call will silently discard entries without a comma. Would it be better to report an error instead?

Also collecting into HashMap will handle duplicate keys in the "last one wins" fashion. Probably not a big deal but maybe consider making that into an error too?

---

The `split(',')` will return an array with the empty string, so will proceed through the chain, but you're right about the `filter_map()` on settings lacking `':'`. This is now fixed.